### PR TITLE
Fix driver compilation for latest kernel

### PR DESCRIPTION
**Pull request by npmccallum**
Warnings still remain for this:

```
/tmp/SGXDataCenterAttestationPrimitives/driver/linux/sgx_main.c: In function ‘sgx_dev_init’:
/tmp/SGXDataCenterAttestationPrimitives/driver/linux/sgx_main.c:395:2: warning: ignoring return value of ‘sysfs_create_file’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  395 |  sysfs_create_file(sgx_dev->kobj_dir, &info_attr.attr);
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/SGXDataCenterAttestationPrimitives/driver/linux/sgx_main.c:396:2: warning: ignoring return value of ‘sysfs_create_file’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  396 |  sysfs_create_file(sgx_dev->kobj_dir, &version_attr.attr);
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

But I'm not a Linux kernel developer, so I'm not quite sure how to handle cleanup of those sysfs entries should one of those commands fail.

### CODE REVIEW

**Comment by haitaohuang** [comment]
we  probably don't want to change the name to avoid breaking existing user script.
Other changes may need revisit after we merge v26+ changes  after next release to avoid conflicts

File: driver/linux/sgx_page_cache.c [hunk 1]
```diff
@@ -64,6 +64,7 @@
 #include <linux/version.h>
 #include <linux/sched/signal.h>
 #include <linux/slab.h>
+#include <linux/io.h>
 #include "sgx.h"
```
Comment (haitaohuang) [inline driver/linux/sgx_page_cache.c]: will not be needed after v26+ changes

File: driver/linux/sgx.h [hunk 2]
```diff
@@ -78,14 +78,14 @@
...
-    #define MSR_IA32_FEATURE_CONTROL 		0x0000003a
+#ifndef MSR_IA32_FEAT_CTL
+    #define MSR_IA32_FEAT_CTL 		0x0000003a
 #endif
 
-#ifndef FEATURE_CONTROL_SGX_LE_WR
-    #define FEATURE_CONTROL_SGX_LE_WR		(1<<17)
+#ifndef FEAT_CTL_SGX_LE_WR
+    #define FEAT_CTL_SGX_LE_WR		(1<<17)
 #endif
```
Comment (haitaohuang) [inline driver/linux/sgx.h]: part of v26+ changes

**Comment by npmccallum** [comment]
@haitaohuang It doesn't encourage external contribution to develop large changes internally and then dump them over the wall. This here is a great example.

Because you're developing a large code-dump privately instead of in the open, my fixes and improvements have to wait until your next development cycle. This creates a two-tier contributor system where Intel's development efforts have a higher priority than external contributors.

If this were just Intel's project, then that would be entirely your prerogative. But the Intel SGX effort is now a Confidential Computing Consortium project where it is expected to be able to receive contributions from all parties equally.

**Comment by spjohnso** [comment]
FYI. This project is not currently a CCC project.

**Comment by npmccallum** [comment]
@spjohnso Thanks for the clarification. I apologize for the misunderstanding.

Regardless, we both want SGX to succeed. Turning away contributions to your open source project is not a good way to encourage that success.

**Comment by spjohnso** [comment]
@npmccallum your point is understood.

**Comment by haitaohuang** [comment]
@npmccallum the v26 change was done, it's just temporarily reverted. See history of commits.
Anyway, not intended to be secret or anything.

**Comment by haitaohuang** [comment]
Also just push v27_pr branch from my repo. Those are all changes planned so far in pipeline. Will update the branch if more changes are needed. You can also rebase this PR with v27_pr

**Comment by haitaohuang** [comment]
changes are merged in different PRs

**Pull request closed without merging**